src/snowflake/cli/_plugins/nativeapp/codegen/templates/templates_processor.py

```diff
@@ -41,16 +42,52 @@ class TemplatesProcessor(ArtifactProcessor):
     Processor class to perform template expansion on all relevant artifacts (specified in the project definition file).
     """
 
+    def expand_templates_in_file(self, src: Path, dest: Path) -> None:
+        """
+        Expand templates in the file.
+        """
+        if src.is_dir():
+            return
+        with self.edit_file(dest) as file:
+            file_name = src.relative_to(self._bundle_ctx.project_root)
+            jinja_env = (
+                choose_sql_jinja_env_based_on_template_syntax(
+                    file.contents, reference_name=file_name
+                )
+                if dest.name.lower().endswith(".sql")
+                else get_client_side_jinja_env()
+            )
+
+            try:
+                expanded_template = jinja_env.from_string(file.contents).render(
+                    get_cli_context().template_context
+                )
+
+            # For now, we are printing the source file path in the error message
+            # instead of the destination file path to make it easier for the user
+            # to identify the file that has the error, and edit the correct file.
+            except jinja2.TemplateSyntaxError as e:
+                raise InvalidTemplateInFileError(file_name, e, e.lineno) from e
+
+            except jinja2.UndefinedError as e:
+                raise InvalidTemplateInFileError(file_name, e) from e
+
+            if expanded_template != file.contents:
+                cc.message(f"Expanding templates in {file_name}")
```

Comment:
too late, since there could be errors raised before this is output. I get that you don't want to say anything about non-templated files, but we should then at least buffer the exceptions until we have a chance to tell the user what step is about to fail.

Reply:
I thought about this, but since all errors will have the file name included, it might not be worth it. WDYT?

Reply:
Definitely worth it IMHO. We should always tell the user what we're doing, and do so accurately.